site/src/sphinx/client-circuit-breaker.rst

```diff
@@ -1,6 +1,273 @@
+.. _CircuitBreaker wiki page: https://martinfowler.com/bliki/CircuitBreaker.html
+.. _LINE Engineering blog post about circuit breaker: https://engineering.linecorp.com/en/blog/detail/76
+
 .. _client-circuit-breaker:
 
-Circuit breakers
-================
+Circuit breaker
+===============
+
+In microservice architecture, it's common that various services running on different machines are connected to
+each other through remote calls. If one of the services becomes unreachable somehow such as due to network
+issues, the client which connects to that service will take long to get a response from it or to fail to
+get one. Situation will get even worse if there are many remote calls to such unresponsive service.
+You can configure an Armeria client with a circuit breaker to prevent this circumstance. The circuit breaker
+can automatically detect failures by continuously updating success and failure events. If the remote service
+is unresponsive, it will immediately respond with an error and not make remote calls.
+Please refer to `CircuitBreaker wiki page`_ by Martin Fowler and
+`LINE Engineering blog post about circuit breaker`_ for more information.
+
+State of ``CircuitBreaker``
+---------------------------
+
+A :api:`CircuitBreaker` can be one of the following three states:
+
+- ``CLOSED``
+
+  - Initial state. All requests are treated normally.
+
+- ``OPEN``
+
+  - The state changes to ``OPEN`` once the number of failures divided by the total number of requests exceeds
+    the failure rate. All requests are blocked off responding with :api:`FailFastException`.
+
+- ``HALF_OPEN``.
+
+  - After a certain amount of time in the ``OPEN`` state, it changes to the ``HALF_OPEN`` state which sends
+    a request to find out if the service is still unavailable or not.
+    If the request succeeds, the state changes to ``CLOSED``. If it fails, the state changes to ``OPEN``.
```

Comment:
- the state changes to CLOSED. -> it enters the CLOSED state.
- the state changes to OPEN. -> it enters the OPEN state again.